### ISSUE TITLE
Building Error (Unity 2019.3.3f1)

### ISSUE THREAD

**Issue by RamoramaInteractive** [googlesamples/arcore-depth-lab#19]
I'm using Unity 2019.3.3f1 Personal. After opening **arcore-depth-lab** and importing **arcore-unity-sdk-1.20.0.unitypackage** I set the version of ARCore Foundation from 2.0.2 to 3.1.3.

But if I build the app I always receiving following errors:

> BuildFailedException: GoogleARCore detected. Google's "ARCore SDK for Unity" and Unity's "ARCore XR Plugin" package cannot be used together. If you have already removed GoogleARCore, you may need to restart the Editor.
> UnityEditor.XR.ARCore.ARCorePreprocessBuild.EnsureGoogleARCoreIsNotPresent () (at Library/PackageCache/com.unity.xr.arcore@3.1.3/Editor/ARCoreBuildProcessor.cs:79)
> UnityEditor.XR.ARCore.ARCorePreprocessBuild.OnPreprocessBuild (UnityEditor.Build.Reporting.BuildReport report) (at Library/PackageCache/com.unity.xr.arcore@3.1.3/Editor/ARCoreBuildProcessor.cs:30)
> UnityEditor.Build.BuildPipelineInterfaces+<>c__DisplayClass15_0.<OnBuildPreProcess>b__1 (UnityEditor.Build.IPreprocessBuildWithReport bpp) (at <9a184ab867bb42c296d20ace04f48df3>:0)
> UnityEditor.Build.BuildPipelineInterfaces.InvokeCallbackInterfacesPair[T1,T2] (System.Collections.Generic.List`1[T] oneInterfaces, System.Action`1[T] invocationOne, System.Collections.Generic.List`1[T] twoInterfaces, System.Action`1[T] invocationTwo, System.Boolean exitOnFailure) (at <9a184ab867bb42c296d20ace04f48df3>:0)
> UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)
> 
> BuildFailedException: 'Preferences > External Tools > Android > Gradle' is empty. ARCore SDK for Unity requires a customized Gradle with version >= 5.6.4.
> GoogleARCoreInternal.ARCoreAndroidSupportPreprocessBuild.CheckGradleVersion () (at Assets/GoogleARCore/SDK/Scripts/Editor/ARCoreAndroidSupportPreprocessBuild.cs:123)
> GoogleARCoreInternal.ARCoreAndroidSupportPreprocessBuild.OnPreprocessBuild (UnityEditor.BuildTarget target, System.String path) (at Assets/GoogleARCore/SDK/Scripts/Editor/ARCoreAndroidSupportPreprocessBuild.cs:59)
> GoogleARCoreInternal.PreprocessBuildBase.OnPreprocessBuild (UnityEditor.Build.Reporting.BuildReport report) (at Assets/GoogleARCore/SDK/Scripts/Editor/PreprocessBuildBase.cs:52)
> UnityEditor.Build.BuildPipelineInterfaces+<>c__DisplayClass15_0.<OnBuildPreProcess>b__1 (UnityEditor.Build.IPreprocessBuildWithReport bpp) (at <9a184ab867bb42c296d20ace04f48df3>:0)
> UnityEditor.Build.BuildPipelineInterfaces.InvokeCallbackInterfacesPair[T1,T2] (System.Collections.Generic.List`1[T] oneInterfaces, System.Action`1[T] invocationOne, System.Collections.Generic.List`1[T] twoInterfaces, System.Action`1[T] invocationTwo, System.Boolean exitOnFailure) (at <9a184ab867bb42c296d20ace04f48df3>:0)
> UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)
> 
> Error building Player: 2 errors
> 
> Build completed with a result of 'Failed'
> UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)
> 
> UnityEditor.BuildPlayerWindow+BuildMethodException: 3 errors
>   at UnityEditor.BuildPlayerWindow+DefaultBuildMethods.BuildPlayer (UnityEditor.BuildPlayerOptions options) [0x00275] in <9a184ab867bb42c296d20ace04f48df3>:0 
>   at UnityEditor.BuildPlayerWindow.CallBuildMethods (System.Boolean askForBuildLocation, UnityEditor.BuildOptions defaultBuildOptions) [0x00080] in <9a184ab867bb42c296d20ace04f48df3>:0 
> UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)
> 

I always receive following Android Resolver message:
![Unbenannt](https://user-images.githubusercontent.com/6389713/97849720-3d89be00-1cf3-11eb-9f93-19eaa8014fce.PNG)

I'm using Android Studio 4.0.1 and I already have jdk-14.0.2. I don't understand the error. 
How to fix the error?

**Comment by ruofeidu** [comment]
Our code base is majorly validated with Unity 2018.4.24f1 and ARCore SDK for Unity v1.18.0.

For the latest Unity and ARCore, please follow https://developers.google.com/ar/develop/unity/android-11-build to configure the gradle etc. I used Unity 2020.1 or later and it worked great for me.

**Comment by RamoramaInteractive** [comment]
I had overseen in the Unity 2019.3 section that the customizing only supports Unity 2019.3 patch 7 and later.

I finally managed to build the app.

Now I've got a new problem. After opening the build App on my Huawei Mate 20 Lite the app quits immedately. Huawei Mate 20 Lite has the OS Oreo (Android 8.1 - API Level 27).

The Depth Lab App from the Google Playstore works but not the built one.

What could be the error? Shall I upgrade AR Foundation or shall I use Unity 2020?

**Comment by RamoramaInteractive** [comment]
I had overseen in the Unity 2019.3 section that the customizing only supports Unity 2019.3 patch 7 and later.
I finally managed to build the app.

Now I've got a new problem. After opening the build App on my Huawei Mate 20 Lite the app quits immedately. Huawei Mate 20 Lite has the OS Oreo (Android 8.1 - API Level 27).

The Depth Lab App from the Google Playstore works but not the built one.

What could be the error?
